pkg/api/addtask.go

```diff
@@ -0,0 +1,99 @@
+package api
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"strconv"
+	"time"
+
+	"github.com/Perehodko/final_project_go/pkg/db"
+)
+
+// addTaskHandler обрабатывает POST запросы для добавления задач
+func addTaskHandler(w http.ResponseWriter, r *http.Request) {
+	if r.Method != http.MethodPost {
+		writeJSONError(w, "Метод не поддерживается", http.StatusMethodNotAllowed)
+		return
+	}
+
+	// Парсим JSON из тела запроса
+	var task db.Task
+	if err := json.NewDecoder(r.Body).Decode(&task); err != nil {
+		writeJSONError(w, "Ошибка разбора JSON: "+err.Error(), http.StatusBadRequest)
+		return
+	}
+
+	// Валидируем и обрабатываем задачу
+	if err := processTask(&task); err != nil {
+		writeJSONError(w, err.Error(), http.StatusBadRequest)
+		return
+	}
+
+	// Добавляем задачу в БД
+	id, err := db.AddTask(&task)
+	if err != nil {
+		writeJSONError(w, err.Error(), http.StatusInternalServerError)
+		return
+	}
+
+	// Возвращаем успешный ответ (конвертируем int64 в string)
+	writeJSONResponse(w, map[string]interface{}{"id": strconv.FormatInt(id, 10)}, http.StatusOK)
+}
+
+// processTask обрабатывает и валидирует данные задачи
+func processTask(task *db.Task) error {
+	// Проверяем обязательное поле title
+	if task.Title == "" {
+		return fmt.Errorf("не указан заголовок задачи")
+	}
+
+	// СНАЧАЛА проверяем правило повторения (если указано)
+	if task.Repeat != "" {
+		// Проверяем правило повторения с помощью NextDate
+		// Используем текущую дату как базовую для проверки
+		now := time.Now()
+		currentDate := now.Format("20060102")
+		_, err := NextDate(now, currentDate, task.Repeat)
+		if err != nil {
+			return fmt.Errorf("неверный формат правила повторения: %v", err)
```

Comment:
При оборачивании нужно использовать `%w` (wrap), а просто при выводе в консоль `%v`
```suggestion
			return fmt.Errorf("неверный формат правила повторения: %w", err)
```